### PR TITLE
Claire work

### PR DESCRIPTION
**Pull request by Clair3**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,7 +12,9 @@
     'scripts/debug.py', 
     'scripts/train.py', 
     'scripts/test.py', 
-    'scripts/tune.py', 
+    'scripts/tune.py',
+    'scripts/analysis.py',
```
Comment (vitusbenson) [inline setup.py:16]: these you did not add to the scripts folder yet

File: earthnet_models_pytorch/model/channel_u_net.py [hunk 2]
```diff
@@ -1,140 +0,0 @@
-"""Channel-U-Net
```
Comment (vitusbenson) [inline earthnet_models_pytorch/model/channel_u_net.py:1]: This file was probably unintentionally deleted, or is there another reason?

File: earthnet_models_pytorch/model/__init__.py [hunk 3]
```diff
@@ -1,24 +1,41 @@
 
 
+from lib2to3.pytree import Base
```
Comment (vitusbenson) [inline earthnet_models_pytorch/model/__init__.py:3]: what is this for?

File: configs/en21x/context-convlstm/concat/base.yaml [hunk 4]
```diff
@@ -0,0 +1,104 @@
+Architecture: "context-convlstm"
```
Comment (vitusbenson) [inline configs/en21x/context-convlstm/concat/base.yaml:1]: probably can remove this, or did you ever work with the old en21x setting?

File: earthnet_models_pytorch/model/local_rnn.py [hunk 5]
```diff
@@ -142,11 +144,12 @@ def forward(self, data, pred_start: int = 0, n_preds: Optional[int] = None):
 
         else:
 
-            meso_dynamic_inputs = meso_dynamic_inputs.reshape(b*t_m, c_m, h_m, w_m)
+            meso_dynamic_inputs = meso_dynamic_inputs.reshape(b*t_m, c_m, 1, 1) #h_m, w_m)
```
Comment (vitusbenson) [inline earthnet_models_pytorch/model/local_rnn.py:147]: we should add this somehow that it both works for en21 (with mesoscale weather) and for en23

File: earthnet_models_pytorch/model/dumby_mlp.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            nn.ReLU(), 
+            nn.Linear(dim_features[2], dim_features[3]),
+            nn.ReLU()
+        )
+
+    def forward(self, weather):
+        return self.model(weather)
+
+
+class DumbyMLP(nn.Module):
```
Comment (vitusbenson) [inline earthnet_models_pytorch/model/dumby_mlp.py:33]: maybe rename this? DumbyMLP is somewhat undescriptive of the model

File: earthnet_models_pytorch/task/loss.py [hunk 7]
```diff
@@ -99,56 +96,52 @@ def forward(self, preds, batch, aux, current_step = None):
...
-        
-        targs = batch["dynamic"][0][:,-preds.shape[1]:,...]
+        targs = batch["dynamic"][0][:,-preds.shape[1]:,...] 
         if len(batch["dynamic_mask"]) > 0:
             masks = batch["dynamic_mask"][0][:,-preds.shape[1]:,...]
         else:
             masks = None
-        
         if self.ndvi:
-            if targs.shape[2] >= 3 and self.comp_ndvi:
```
Comment (vitusbenson) [inline earthnet_models_pytorch/task/loss.py:124]: your editing in this file is not backwards compatible, but i guess it is fine since you want to redo the whole loss API anyway

File: earthnet_models_pytorch/utils/logging.py [hunk 8]
```diff
@@ -7,6 +7,7 @@
 import numpy as np
 from PIL import Image, ImageDraw, ImageFont
 import matplotlib.colors as clr
+import sys
```
Comment (vitusbenson) [inline earthnet_models_pytorch/utils/logging.py:10]: you dont seem to use this, unnecessary import?

File: earthnet_models_pytorch/utils/parse.py [hunk 9]
```diff
@@ -82,32 +84,48 @@ def parse_setting(setting_file, track = None):
...
             lr_per_sample = optimizer["lr_per_sample"]
-            lr = bs * (gpus * ddp + (1-ddp)) * lr_per_sample
+            if isinstance(gpus, list):
+                lr = bs * (len(gpus) * ddp + (1-ddp)) * lr_per_sample
+            else:    
+                lr = bs * (gpus * ddp + (1-ddp)) * lr_per_sample
+            print('learning rate', lr)
             optimizer["args"]["lr"] = lr
         
-    if track is not None:
```
Comment (vitusbenson) [inline earthnet_models_pytorch/utils/parse.py:100]: please keep this part. The idea is to set the correct context and target lengths for a given setting. We can change this later as part of the refactoring.

**Review by vitusbenson — CHANGES_REQUESTED** [review]
despite the minor comments i did, looks good! :)
Once you changed those, I will merge.

File: earthnet_models_pytorch/model/channel_u_net.py (hunk 2 above)
Comment (Clair3) [reply earthnet_models_pytorch/model/channel_u_net.py:1]: Yes, it is useless, I tried some things, but it was not great.

File: earthnet_models_pytorch/task/loss.py (hunk 7 above)
Comment (Clair3) [reply earthnet_models_pytorch/task/loss.py:124]: yes, it is unclear, I actually didn't know what exactly you want to encode by self.ndvi, I think it didn't work on en22 with this if, I remove it instead to add a condition because self.comp_ndvi is unclear neither, and I tried to only run a model . What do you think if we change the condition on the shape on a condition to the dataset? And compute the target in the dataloader file avoid having calculations in several places and having dimensions that vary according to the target?

**Review by vitusbenson — APPROVED** [review]
